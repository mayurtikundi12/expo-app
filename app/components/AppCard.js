import { View, Image, StyleSheet, TouchableHighlight, Button } from 'react-native';
// import Swipeable from 'react-native-gesture-handler/Swipeable';

import React from 'react';

import AppColours from '../config/AppColours';
import AppText from './AppText';

function AppCard({title, subtitle, image, onPress}) {
    return (
        // <Swipeable renderRightActions={onSwipeLeft}>    
            <TouchableHighlight onPress={onPress} underlayColor={AppColours.white}>   
                <View style={styles.container}>
                    <Image source={image} style={styles.image}/>
                    <AppText style={styles.title}> {title} </AppText>
                    <AppText style={styles.subtitle}> {subtitle} </AppText>
                </View>
            </TouchableHighlight> 
        // </Swipeable>
    );
}
const styles = StyleSheet.create({
    container:{
        backgroundColor: AppColours.lightG,
        borderRadius: 20,
        overflow: "hidden",
        marginBottom: 25,
    },
    image: {
        height: 200,
        width: "100%",
    },
    title: {
        fontSize: 20,
        color: AppColours.white,
        fontFamily: "Roboto",
        alignItems: 'center',
        fontWeight: 'bold',
    },
    subtitle: {
        fontSize: 15,
        color: AppColours.white,
        fontFamily: "Roboto",
        alignItems: 'center',
    },
})

export default AppCard;