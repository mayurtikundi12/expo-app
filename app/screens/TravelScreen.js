import React ,{useState, useEffect}from 'react';
import { ScrollView, StyleSheet, View ,Button} from 'react-native';
import AppCard from '../components/AppCard';
import AppColours from '../config/AppColours';
import AppText from '../components/AppText';


function TravelScreen(props) {
    let placeList = [{
        title:"Bora Bora",
        subtitle:"Undoubtedly the most celebrated island in the South Pacific, Bora Bora is French Polynesia's leading lady. ",
        image:require("../../assets/bora-bora.jpg")
    },
    {                   
    title:"Switzerland",
    subtitle:"Switzerland, officially the Swiss Confederation, is a landlocked country situated at the confluence of Western, Central, and Southern Europe.",
    image:require("../../assets/switz.jpg")
    },
    {
    title:"Mykonos",
    subtitle:"Mykonos is a Greek island, part of the Cyclades, lying between Tinos, Syros, Paros and Naxos.",
    image:require("../../assets/myk.jpg")
    }
];

    let extraPlaces = [{     
        title:"Taj Mahal",
        subtitle:"One of the 7 Wonder's of the world",
        image:require("../../assets/bora-bora.jpg")
}]
    function showMore(){

        if(extraPlaces.length > 0){
        let newPlace = extraPlaces.pop();
        setPlaces([...places,newPlace]);
    }
    }
    let [places,setPlaces] = useState(placeList);
    useEffect(()=>{

    },[places])
    return (
        <ScrollView pagingEnabled={true} >
        <View style={styles.container}>
          
            <AppText style={styles.apptext}>
                Select your travel destination!
            </AppText>
            {
                places.map((item,index)=>{
                  return  <AppCard  key={index} testId="placeCard" style={styles.card}
                    title={item.title}
                    subtitle={item.subtitle}
                    image={item.image}
                />
                })
            }
        </View>
        <Button testID="showMore" title="Show More Places" onPress={showMore} >
        </Button>
        </ScrollView>
    );
}
const styles = StyleSheet.create({
    container: {
        flex:1,
        backgroundColor: AppColours.darkG,
    },
    card:{
        zIndex:5,
        padding: 10,
        marginTop: 10,
    },  
    apptext: {
        fontSize: 25,
        fontFamily: "Roboto",
        alignItems: 'center',
        fontWeight: 'bold',
        padding: 10,
        color: AppColours.white,
    },
})

export default TravelScreen;