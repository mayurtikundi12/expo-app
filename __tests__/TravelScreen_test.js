import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import TravelScreen from '../app/screens/TravelScreen.js';
import {act} from '@testing-library/react-native';
import AppCard from '../app/components/AppCard.js';
describe('Travel Screen test', () => {
    let tree, treeInstance;
    beforeAll(async () => {
       tree = await renderer.create( 
            <TravelScreen/> 
        )
        treeInstance = tree.root
    });

    it('Account Screen Test', async () => {
        expect(tree.toJSON()).toMatchSnapshot()
    })

    it('Update Cards List', async() => {
        let showMoreButton = treeInstance.findByProps({testID: 'showMore'});
        expect(showMoreButton).toBeTruthy();
        let cards = treeInstance.findAllByType(AppCard);
        expect(cards).toBeTruthy()
        let originalLength = cards.length;
        await act(async()=>{
            await showMoreButton.props.onPress();
        })
        let updatedCards = treeInstance.findAllByType(AppCard);
        expect(updatedCards.length).toEqual(originalLength+1);
    })

})


