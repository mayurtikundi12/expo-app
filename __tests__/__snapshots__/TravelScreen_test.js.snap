// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel Screen test Account Screen Test 1`] = `
<RCTScrollView
  pagingEnabled={true}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#071e22",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Optima",
              "fontSize": 20,
            },
            Object {
              "alignItems": "center",
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
         
        Select your travel destination!
         
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#1d7874",
              "borderRadius": 20,
              "marginBottom": 25,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 200,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
             
             
            Bora Bora
             
             
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                },
              ]
            }
          >
             
             
            Undoubtedly the most celebrated island in the South Pacific, Bora Bora is French Polynesia's leading lady. 
             
             
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#1d7874",
              "borderRadius": 20,
              "marginBottom": 25,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 200,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
             
             
            Switzerland
             
             
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                },
              ]
            }
          >
             
             
            Switzerland, officially the Swiss Confederation, is a landlocked country situated at the confluence of Western, Central, and Southern Europe.
             
             
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#1d7874",
              "borderRadius": 20,
              "marginBottom": 25,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 200,
                "width": "100%",
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
             
             
            Mykonos
             
             
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "color": "#fff",
                  "fontFamily": "Roboto",
                  "fontSize": 15,
                },
              ]
            }
          >
             
             
            Mykonos is a Greek island, part of the Cyclades, lying between Tinos, Syros, Paros and Naxos.
             
             
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="showMore"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Show More Places
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
